File: 卡码/代码随想录/1/08-贪心算法/Go/02-中等/LeetCode_0045.go
// 45. 跳跃游戏 II
package main

import "math"

func jump(nums []int) int {
	res,cur,next:=0,0,0
	for i:=0;i<len(nums)-1;i++ {
		next=int(math.Max(float64(i+nums[i]),float64(next)))
		if i==cur {
			cur=next
			res++
		}
	}
	return res
}

File: 卡码/代码随想录/1/08-贪心算法/Js/02-中等/LeetCode_0045.js
//45. 跳跃游戏 II
const jump = function(nums) {
  let res=0,cur=0,next=0;
  for(let i=0;i<nums.length-1;i++){
    next=Math.max(next,i+nums[i]);
    if(i==cur){
      cur=next;
      res++;
    }
  }
  return res;
};